include/libhal-icm/icm20948.hpp

```diff
@@ -187,85 +184,93 @@ typedef enum ICM20948_CYCLE
    */
   [[nodiscard]] hal::result<mag_read_t> read_magnetometer();
 
-
   /**
    * @brief Read pressure data from out_t_msb_r and out_t_lsb_r
    *        and perform temperature conversion to celsius.
    */
   [[nodiscard]] hal::result<temp_read_t> read_temperature();
 
-
-
-  [[nodiscard]] static result<icm20948> create(hal::i2c& p_i2c, hal::byte p_device_address);
+  [[nodiscard]] static result<icm20948> create(hal::i2c& p_i2c);
 
   hal::status init();
+
   hal::status auto_offsets();
-  hal::status set_acc_offsets(float p_xMin,
-                            float p_xMax,
-                            float p_yMin,
-                            float p_yMax,
-                            float p_zMin,
-                            float p_zMax);
-  hal::status set_gyr_offsets(float p_xOffset, float p_yOffset, float p_zOffset);
+
+  /**
+   * @brief Set default acceleration offsets
+   *    All offset parameters are in 'g's
+   */
+  hal::status set_acc_offsets(float p_xmin,
+                              float p_xmax,
+                              float p_ymin,
+                              float p_ymax,
+                              float p_zmin,
+                              float p_zmax);
+
+  hal::status set_gyro_offsets(float p_x_offset,
+                               float p_y_offset,
+                               float p_z_offset);
+
   hal::result<hal::byte> whoami();
 
-  hal::status enable_acc(bool p_enAcc);
-  hal::status set_acc_range(ICM20948_accRange p_accRange);
-  hal::status set_acc_DLPF(ICM20948_dlpf p_dlpf);
-  hal::status set_acc_sample_rate_div(uint16_t p_accSplRateDiv);
+  hal::status enable_acc(bool p_en_acc);
+  hal::status set_acc_range(icm20948_acc_range p_acc_range);
+  hal::status set_acc_dlpf(icm20948_dlpf p_dlpf);
```

Comment:
```suggestion
  hal::status set_acc_range(const icm20948_acc_range& p_acc_range);
  hal::status set_acc_dlpf(const icm20948_dlpf& p_dlpf);
```

Lets also remove the `icm20948_` prefix as well. And what is the `dlpf`? I have no clue at all. Can you spell it out or document what it means? These APIs should be documented using the doxygen style. See `libhal/libhal`'s header's files for an example. You can also take a look at `libhal-armcortex` as well.

Do the same for the rest.